### ISSUE TITLE
ENH: Support reading empty DataFrame via max_features=0

### ISSUE THREAD

**Issue by brendan-ward** [geopandas/pyogrio#44]
Right now `max_features=0` denotes unset / read all features.  Change this so that it allows populating an empty DataFrame.

**Comment by Imanflow** [comment]
Out of the issues if seen around, this seems the easiest, and the one I might be able to tackle. But could you give some further explanation? 

My doubts: 

Do you mean reading an empty .geojson file as an empty DF?

**Comment by brendan-ward** [comment]
No, I mean construct an empty data frame from a (likely) non-empty dataset.

A little more info [here](https://github.com/geopandas/dask-geopandas/pull/123#discussion_r803957165)

This requires changes in Cython...  I think we'd want to use -1 as the sentinal value to indicate read all records (so `validate_feature_range` will need to be updated to allow that), [this line](https://github.com/geopandas/pyogrio/blob/main/pyogrio/_io.pyx#L552-L553) will need to be changed to `max_features >= 0`, default values will need to be updated, and tests added...

**Comment by Imanflow** [comment]
I was afraid this library would be too low lever for me to help, and indeed I think it's the case. I've never done anything in Cython.
